Panic when encoding empty slice with "comma" tag.

Running this code results in panic:

```go
package main

import (
	"fmt"

	"github.com/sonh/qs"
)

type Query struct {
	Tags []string `qs:"tags,comma"`
}

func main() {
	encoder := qs.NewEncoder()

	values, _ := encoder.Values(&Query{Tags: []string{}})
	fmt.Println(values.Encode())
}

```

Output

```
panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
github.com/sonh/qs.(*listField).formatFnc(0xc00007c080, {0x4a5140?, 0xc000010048?, 0xc000014280?}, 0xc000014290)
        /home/pnx/go/pkg/mod/github.com/sonh/qs@v0.6.0/encode_field.go:116 +0x2f5
github.com/sonh/qs.(*encoder).encodeStruct(0xc00009c040, {0x4ab780?, 0xc000010048?, 0x40c7e7?}, 0xc0000921b0, {0x0, 0x0, 0x0})
        /home/pnx/go/pkg/mod/github.com/sonh/qs@v0.6.0/encode.go:190 +0x391
github.com/sonh/qs.(*Encoder).Values(0xc00007c020, {0x4a3400?, 0xc000010048?})
        /home/pnx/go/pkg/mod/github.com/sonh/qs@v0.6.0/encode.go:94 +0x1f4
main.main()
        /home/pnx/code/qs/test.go:16 +0x54
exit status 2
```

Correct behavior would be to encode the list into a empty string.

--- a/encode_field.go
+++ b/encode_field.go
@@ -113,7 +113,7 @@ func (listField *listField) formatFnc(field reflect.Value, result resultFunc) er
 			}
 		}
 		returnStr := str.String()
-		if returnStr[0] == ',' {
+		if len(returnStr) > 0 && returnStr[0] == ',' {
 			returnStr = returnStr[1:]
 		}
 		result(listField.name, returnStr)

--- a/encode_test.go
+++ b/encode_test.go
@@ -602,6 +602,7 @@ func TestArrayFormat_Comma(t *testing.T) {
 	tm := time.Unix(600, 0).UTC()
 
 	s := struct {
+		EmptyList  []string     `qs:"empty_list,comma"`
 		StringList []string     `qs:"str_list,comma"`
 		Times      []*time.Time `qs:"times,comma"`
 	}{
@@ -614,8 +615,9 @@ func TestArrayFormat_Comma(t *testing.T) {
 		return
 	}
 	expected := url.Values{
-		"str_list": []string{"a,b,c"},
-		"times":    []string{tm.Format(time.RFC3339)},
+		"empty_list": []string{""},
+		"str_list":   []string{"a,b,c"},
+		"times":      []string{tm.Format(time.RFC3339)},
 	}
 	assert.Equal(t, expected, values)
 }

